src/Nethermind/Nethermind.JsonRpc/Modules/Trace/TraceRpcModule.cs

```diff
@@ -110,46 +118,19 @@ public ResultWrapper<ParityTxTraceFromReplay> trace_rawTransaction(byte[] data,
             return TraceTx(tx, traceTypes, BlockParameter.Latest);
         }
 
-        private SearchResult<BlockHeader> SearchBlockHeaderForTraceCall(BlockParameter blockParameter)
+        private ResultWrapper<ParityTxTraceFromReplay> TraceTx(Transaction tx, string[] traceTypes, BlockParameter blockParameter,
+            Dictionary<Address, AccountOverride>? stateOverride = null)
         {
             SearchResult<BlockHeader> headerSearch = _blockFinder.SearchForHeader(blockParameter);
             if (headerSearch.IsError)
-            {
-                return headerSearch;
-            }
-
-            BlockHeader header = headerSearch.Object;
-            if (header!.IsGenesis)
-            {
-                UInt256 baseFee = header.BaseFeePerGas;
-                header = new BlockHeader(
-                    header.Hash!,
-                    Keccak.OfAnEmptySequenceRlp,
-                    Address.Zero,
-                    header.Difficulty,
-                    header.Number + 1,
-                    header.GasLimit,
-                    header.Timestamp + 1,
-                    header.ExtraData,
-                    header.BlobGasUsed,
-                    header.ExcessBlobGas);
-
-                header.TotalDifficulty = 2 * header.Difficulty;
-                header.BaseFeePerGas = baseFee;
-            }
-
-            return new SearchResult<BlockHeader>(header);
```

Comment:
Is this now not needed?

Reply:
As I understood, It was used to fix issues with rollback to the previous block being applied on the genesis - for `trace_call`, `trace_callMany` and `trace_rawTransaction`. Not needed as I removed this rollback for those methods.